notify/receiver/receiver.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receiver
```

Comment:
This had to be moved to a new package due to package import cycles.

Happy to choose any other location for this.

Reply:
Fun fact - this function actually did used to be exported, up until the notify package was split up.

There's a relevant comment here: https://github.com/prometheus/alertmanager/pull/1929#discussion_r295179478

Reply:
I like how every package inside `notify` (with a very obvious exception in `test`) is an integration. Can this not be part of the `notify` package or a completely separate package?

Reply:
Can't be part of `notify` due to package dependencies. All the integrations depend on `notify`, and then this code depends on all the integrations, so this has to be its own package.

I moved it to a subpackage of `config` instead (`config/receiver`), since it's logically glue code between config and notify. WDYT?

I can't make it part of`config` directly for the same reason as notify - import cycles. It needs to be a subpackage somewhere.

Reply:
Yeah, I'm happy to keep this in config.